model/src/main/kotlin/spdx/SpdxExpression.kt

```diff
@@ -31,20 +31,43 @@ data class SpdxCompoundExpression(
         val left: SpdxExpression,
         val operator: SpdxOperator,
         val right: SpdxExpression
-) : SpdxExpression()
+) : SpdxExpression() {
+    override fun toString(): String {
+        val leftString = when {
+            left is SpdxCompoundExpression && operator.priority > left.operator.priority -> "($left)"
+            else -> "$left"
+        }
+
+        val rightString = when {
+            right is SpdxCompoundExpression && operator.priority >= right.operator.priority -> "($right)"
```

Comment:
Why is it `>=` here but `>` above?

Reply:
This is not required to keep the semantics of the expression, but to keep the structure. For example `license1 AND license2 AND license3` is parsed to `(license1 AND license2) AND license3`, but has the same meaning as `license1 AND (license2 AND license3)`. With the `>=` we keep `license1 AND (license2 AND license3)` as is, with only `=` we would serialize it to `license1 AND license2 AND license3`.
But I'm not so sure about this anymore, especially since I have a test to check that unnecessary parenthesis are removed. So I tend to change that, what do you think?

Reply:
I guess by `=` you mean `>`? Anyway, I'd prefer `license1 AND license2 AND license3` over `license1 AND (license2 AND license3)`.

Reply:
Done, also added a code comment to explain this.